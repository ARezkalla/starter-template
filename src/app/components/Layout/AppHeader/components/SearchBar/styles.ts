import { useTheme } from "@mui/material"
const DEFAULT_SPACING_FACTOR = 8

export const styles = (theme = useTheme()) => ({
  search: {
    position: "relative",
    borderRadius: "4px",
    backgroundColor: "background.default",
    marginRight: theme.spacing(2),
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      marginLeft: theme.spacing(3),
      width: "auto",
    },
  },
  iconWrapper: {
    padding: theme.spacing(0, 2),
    height: "100%",
    position: "absolute",
    right: 0,
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    zIndex: 2,
  },
  inputBase: {
    color: "text.secondary",
    backgroundColor: "background.default",
    fontSize: "1rem",
    fontWeight: "400",
    borderRadius: "4px",
    lineHeight: "0.875rem",
    maxHeight: 4 * DEFAULT_SPACING_FACTOR,

    "& .MuiInputBase-input": {
      padding: theme.spacing(1, 1, 1, 0),
      paddingLeft: `calc(1em + ${theme.spacing(4)})`,
      transition: theme.transitions.create("width"),
      width: "100%",
      [theme.breakpoints.up("md")]: {
        width: 75 * DEFAULT_SPACING_FACTOR,
      },
    },
  },
  icon: {
    color: "text.secondary",
  },
})
