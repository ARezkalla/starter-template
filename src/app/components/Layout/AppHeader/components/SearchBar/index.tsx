import { Box } from "@mui/material"
import InputBase from "@mui/material/InputBase"
import SearchIcon from "@mui/icons-material/Search"

import { styles } from "./styles"

const HeaderSearchBar = () => {
  return (
    <Box sx={styles().search}>
      <Box sx={styles().iconWrapper}>
        <SearchIcon sx={styles().icon} />
      </Box>
      <InputBase
        sx={styles().inputBase}
        placeholder={"Search"}
        inputProps={{ "aria-label": "search" }}
      />
    </Box>
  )
}

export default HeaderSearchBar
