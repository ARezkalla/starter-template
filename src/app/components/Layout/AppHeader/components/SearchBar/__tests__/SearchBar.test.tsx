import { renderWithProviders } from "@/app/test/renderWithProvider"
import { describe, expect, it } from "vitest"
import SearchBar from ".."
import "@testing-library/jest-dom"

describe("SearchBar", () => {
  it("should render without crashing", () => {
    const { container } = renderWithProviders(<SearchBar />)
    expect(container).toMatchSnapshot()
  })
})
