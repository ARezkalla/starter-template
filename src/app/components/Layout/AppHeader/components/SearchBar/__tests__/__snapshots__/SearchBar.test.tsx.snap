// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > should render without crashing 1`] = `
<div>
  <div
    class="MuiBox-root css-1wal5io"
  >
    <div
      class="MuiBox-root css-3g3bdw"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18m7kj6-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <div
      class="MuiInputBase-root MuiInputBase-colorPrimary css-h0byw8-MuiInputBase-root"
    >
      <input
        aria-label="search"
        class="MuiInputBase-input css-1u71ta7-MuiInputBase-input"
        placeholder="Search"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
